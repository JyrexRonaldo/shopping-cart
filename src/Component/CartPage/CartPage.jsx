import { useOutletContext } from "react-router-dom";
import CartItemCard from "../CartItemCard/CartItemCard";
import styles from "./CartPage.module.css";

function CartPage() {
  const { productsData, cartItems } = useOutletContext();

  // console.log({productsData, cartItems})

  console.log(cartItems);

  let cartItemCards = [];
  let grandTotal = 0;

  for (const key in cartItems) {
    if (Object.prototype.hasOwnProperty.call(cartItems, key)) {
      // cartItemCards += key
      const itemInfo = productsData.find((product) => {
        return +key === product.id;
      });
      console.log(itemInfo);
      grandTotal += itemInfo.price * cartItems[key];
      cartItemCards.push(
        <CartItemCard
          imgURL={itemInfo.image}
          title={itemInfo.title}
          price={itemInfo.price}
          quantity={cartItems[key]}
        />
      );
    }
  }

  console.log(cartItemCards);

  return (
    <div className={styles.cartPage}>
      <div>
        <h1>Shopping Cart</h1>
        <div>{cartItemCards}</div>
      </div>
      <div>
        Grand Total:{" "}
        {`$${Math.round((grandTotal + Number.EPSILON) * 100) / 100}`}
      </div>
    </div>
  );
}

export default CartPage;
