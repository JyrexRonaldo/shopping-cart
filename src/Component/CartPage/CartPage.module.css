.cartPage {
  background-color: #ebeded;
  flex: 1;
  display: flex;
  gap: 30px;
  justify-content: center;
  align-items: flex-start;
  padding: 40px 200px;
}

.cartPage h1 {
  display: flex;
  justify-content: center;
}

.cartPage > div {
  display: flex;
  flex-direction: column;
  gap: 10px;
  background-color: white;
}

.cartPage > div:nth-child(1) {
  flex: 1 0 250px;
  align-items: center;
  padding: 30px 0px;
}

.cartPage > div:nth-child(1) > div {
  display: flex;
  flex-direction: column;
  gap: 20px;
}

.cartPage > div:nth-child(2) {
  width: 300px;
  justify-content: center;
  align-items: center;
  padding: 20px 40px;
  
}

.cartPage > div:nth-child(2) > p {
  font-weight: bolder;
  font-size: 1.8rem;
}

.cartPage > div:nth-child(2) > button {
  padding: 2px 44px;
}
