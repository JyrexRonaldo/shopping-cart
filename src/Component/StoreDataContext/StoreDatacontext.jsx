import { createContext } from "react";

const StoreDataContext = createContext(null);

export default StoreDataContext