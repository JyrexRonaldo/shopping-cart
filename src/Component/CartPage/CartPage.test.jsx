import { describe, expect, it } from "vitest";
import CartPage from "./CartPage";
import { render, screen } from "@testing-library/react";
import StoreDataContext from "../StoreDataContext/StoreDataContext";

describe("CartPage component", () => {
  it("displays elements as expected", () => {
    const testProductData = {};

    const { container } = render(
      <StoreDataContext.Provider value={testProductData}>
        <CartPage />
      </StoreDataContext.Provider>
    );

    expect(container).toMatchSnapshot();
  });

  it("displays cart items total price when data is available", () => {
    const productsData = [
      
      {
          id: 1,
          title: "Fjallraven - Foldsack No. 1 Backpack, Fits 15 Laptops",
          price: 109.95,
          description:
            "Your perfect pack for everyday use and walks in the forest. Stash your laptop (up to 15 inches) in the padded sleeve, your everyday",
          category: "men's clothing",
          image: "https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg",
          rating: {
            rate: 3.9,
            count: 120,
          },
        },
    
    ];

    const cartItems = {
      1: 1,
    };

    const testProductData = {productsData, cartItems};

    render(
      <StoreDataContext.Provider value={testProductData}>
        <CartPage />
      </StoreDataContext.Provider>
    );

    const paragraghs = screen.getAllByRole('paragraph')

    expect((paragraghs[paragraghs.length - 1]).textContent).toBe("Grand Total: $109.95")
  });
});
