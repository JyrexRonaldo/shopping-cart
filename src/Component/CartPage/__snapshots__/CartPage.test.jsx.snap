// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CartPage component > displays elements as expected 1`] = `
<div>
  <div
    class="_cartPage_848a55"
  >
    <div>
      <h1>
        Shopping Cart
      </h1>
      <div />
    </div>
    <div>
      <p>
        Grand Total:
         
        $0
      </p>
      <button
        type="button"
      >
        Proceed to checkout
      </button>
    </div>
  </div>
</div>
`;
